sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py

```diff
@@ -143,21 +143,17 @@ def _cloud_event_to_generated(cloud_event, **kwargs):
 
 def _from_cncf_events(event):
     """This takes in a CNCF cloudevent and returns a dictionary.
-
-    :raises: AttributeError when a dictionary is passed/ when data attribute is not
-     present in the event.
+    If cloud events library is not installed, the event is returned back.
     """
     try:
         from cloudevents.http import to_json
         return json.loads(to_json(event))
-    except AttributeError:
+    except (AttributeError, ImportError):
         # means this is not a CNCF event
         return event
-    except ImportError:
-        # this means it was not meant to be a cloud event
-        return None
     except: # pylint: disable=bare-except
-        raise ValueError('Failed to deserialize the event.')
+        raise ValueError('Failed to serialize the event. Please ensure your' +
```

Comment:
We should probably use raise_with_traceback to keep the context of what happened:
https://github.com/Azure/azure-sdk-for-python/blob/71bd202f5261e7ce8b2db38bb9b6802f1a55a5dc/sdk/eventgrid/azure-eventgrid/azure/eventgrid/_messaging_shared.py#L27